### ISSUE TITLE
feat[dev]: ✨ implement A11y improvements to sd-chip

### ISSUE THREAD

**Issue by yoezlem** [solid-design-system/solid#1502]
## Description
This task describes the Storybook implementation for the sd-chip component as part of the A11y improvements outlined in Epic https://github.com/solid-design-system/solid/issues/1465.

A11y improvements for sd-chip can be found here: [sd-chip Notes](https://www.notion.so/matuzo/sd-chip-111f3f6e1bc0805fafccd84b0dc8c4a3).

---------------------------------------

## Comment
- [ ] check the note below and fix the templates (seems like the 2 examples in the template are different) => increase story points if it becomes more complex

## Notes
- The heading in a teaser should be the first item in the DOM. A heading introduces a new thematic region and separates the following content from the previous region. Although chips and flags are visually the first items in the teaser, they should come after the heading in the DOM.

## Related WCAG criteria
https://www.w3.org/WAI/WCAG22/quickref/#meaningful-sequence

## DoR
- [x] Item has been estimated by the team
- [x] Item dependencies have been identified and documented

## DoD
- [ ] Documentation has been created/updated (if applicable)
- [ ] Migration Guide has been created/updated (if applicable)
- [ ] Relevant E2E tests (Features, A11y, Bug fixes) are created/updated
- [ ] Relevant stories (Features, A11y) are created/updated
- [ ] Implementation works successfully on `feature` branch
